vignettes/articles/scripting-tasks-config.Rmd

```diff
@@ -0,0 +1,387 @@
+will be.
+
+### Creating the `task_id` objects
+
+In this modeling effort, we want to **require** modelers to provide week-ahead
+predictions for the "weekly incident influenza hospitalizations" target
+(`inc hosp`). Model submissions can optionally include five states, provide up
+to 4-week-ahead predictions, and also provide predictions for the "weekly
+incident influenza deaths" target (`inc death`).
+
+```{r set-task-ids}
+origin_date <- create_task_id(
+  "origin_date",
+  required = NULL,
+  optional = c("2023-01-02", "2023-01-09", "2023-01-16")
+)
+
+location <- create_task_id(
+  "location",
+  required = "US",
+  optional = c("01", "02", "04", "05", "06")
+)
+
+horizon <- create_task_id("horizon", required = 1L, optional = 2:4)
+
+target <- create_task_id("target", required = "inc hosp", optional = "inc death")
+
+task_ids_example <- create_task_ids(origin_date, location, horizon, target)
+
+```
+
+Now that we've created the task IDs that specify the modeling tasks, we can define the outputs we
+expect from the models. 
+
+### Creating the `output_type` objects
+
+For this example, we want to have three output type objects:
+
+1. a required `"mean"` output type with a "double" value
+2. a required `"quantile"` output type with a "double" value
+3. an optional `"median"` output type with an "integer" value
+
+Additionally, our two targets will accept a different combination of output
+types. Specifically, target `"inc hosp"` will only accept `"mean"` and
+`"median"` output types while `"inc death"` will only accept `"mean"` and
+`"quantile"` output types. 
+
+```{r create-output-type-ids}
+mean_out_type <- create_output_type_mean(
+  is_required = TRUE,
+  value_type = "double",
+  value_minimum = 0
+)
+
+median_out_type <- create_output_type_median(
+  is_required = FALSE,
+  value_type = "integer",
+  value_minimum = 0L
+
+)
+
+quantile_out_type <- create_output_type_quantile(
+  required = c(0.25, 0.5, 0.75),
+  is_required = TRUE,
+  value_type = "double",
+  value_minimum = 0
+)
+```
+
+Now that we have our base `output_type_item` class objects, we can combine them to create `output_type` class objects that we want to use for each particular target. 
+
+```{r combine-output-types}
+output_type_mean_median <- create_output_type(mean_out_type, median_out_type)
+
+output_type_mean_quantile <- create_output_type(mean_out_type, quantile_out_type)
+```
+
+### Creating the `model_task` objects
+
+As previously discussed, we want to require a "mean" output for both tasks, but we want a "quantile"
+output type for the `inc death` target and an optional "median" for the `inc
+hosp` target.
+
+```{r create-model-task}
+model_task_hosp <- create_model_task(
+  task_ids = task_ids_example,
+  output_type = output_type_mean_median,
+  target_metadata = target_metadata
+)
+
+model_task_death <- create_model_task(
+  task_ids = task_ids_example,
+  output_type = output_type_mean_quantile,
+  target_metadata = target_metadata
+)
```

Comment:
I'm confused here because In the text just above we said we were going to have different output types for the different targets, but:
- both `model_task_hosp` and `model_task_death` use `task_ids_example`, which included both `target`s
- similarly, it seems like it'd make sense to include only the relevant `target_metadata` for each target

Reply:
This was pretty much my question and I don't know the answer to it: 

> Specifically, why would we have two modeling tasks that have identical task IDs but share only a "mean" output type?


Again, I was adapting this from the tutorial that currently exists.

Reply:
OK, it makes sense to me to merge this in in its currently slightly awkward state, with plans to revisit the example to make it more realistic and self-consistent soon